test/Shared/UserManagerTestBase.cs

```diff
@@ -139,18 +134,17 @@ protected TRole CreateRole(string namePrefix = "")
         public async Task CanUpdatePasswordUsingHasher()
         {
             var manager = CreateManager();
-            var user = new TUser() { UserName = "UpdatePassword" };
+            var user = CreateTestUser("UpdatePassword");
             IdentityResultAssert.IsSuccess(await manager.CreateAsync(user, "password"));
             Assert.True(await manager.CheckPasswordAsync(user, "password"));
-            string expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", user.Id.ToString(), true.ToString());
+            string expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", await manager.GetUserIdAsync(user), true.ToString());
             IdentityResultAssert.VerifyLogMessage(manager.Logger, expectedLog);
 
-            user.PasswordHash = manager.PasswordHasher.HashPassword(user, "New");
-            IdentityResultAssert.IsSuccess(await manager.UpdateAsync(user));
+            IdentityResultAssert.IsSuccess(await manager.ChangePasswordAsync(user, "password", "Password1!"));
             Assert.False(await manager.CheckPasswordAsync(user, "password"));
-            expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", user.Id.ToString(), false.ToString());
+            expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", await manager.GetUserIdAsync(user), false.ToString());
             IdentityResultAssert.VerifyLogMessage(manager.Logger, expectedLog);
-            Assert.True(await manager.CheckPasswordAsync(user, "New"));
+            Assert.True(await manager.CheckPasswordAsync(user, "Password1!"));
```

Comment:
No I replaced the `New` with `Password1!` since it doesn't meet the password requirement. Instead of directly setting the password hash I am doing it via `ChangePasswordAsync`